test/e2e/network/dns_common.go

```diff
@@ -277,38 +281,70 @@ func (t *dnsTestCommon) deleteCoreDNSPods() {
 	}
 }
 
-func generateDNSServerPod(aRecords map[string]string) *v1.Pod {
-	pod := &v1.Pod{
+func generateCoreDNSServerPod(corednsConfig *v1.ConfigMap) *v1.Pod {
+	return &v1.Pod{
 		TypeMeta: metav1.TypeMeta{
 			Kind: "Pod",
 		},
 		ObjectMeta: metav1.ObjectMeta{
 			GenerateName: "e2e-dns-configmap-dns-server-",
 		},
 		Spec: v1.PodSpec{
+			Volumes: []v1.Volume{
+				{
+					Name: "coredns-config",
+					VolumeSource: v1.VolumeSource{
+						ConfigMap: &v1.ConfigMapVolumeSource{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: corednsConfig.Name,
+							},
+						},
+					},
+				},
+			},
 			Containers: []v1.Container{
 				{
 					Name:  "dns",
-					Image: imageutils.GetE2EImage(imageutils.Dnsutils),
+					Image: imageutils.GetE2EImage(imageutils.Agnhost),
```

Comment:
@claudiubelu - the PR you referenced https://github.com/kubernetes/kubernetes/pull/77861/files adds /coredns to dnsutils, not agnhost. Should this be dnsutils still?

Reply:
Other than that the rest makes sense to me but I'd like to get someone from SIG-Networking for LGTM since I'm not an owner on this file

Reply:
Indeed. I've changed that, because I was thinking / hoping that Centralization part 4 PR would merge, which also includes CoreDNS (since it also centralizes dnsutils).